File: frontend/types/Product.ts
export interface Product {
  id: number
  
  main_category: string
  title: string
  average_rating: number
  rating_number: number
  features: string[]
  description: string[]
  price: number
  images: {
    hi_res: string
    thumb: string
    large: string
    variant: string
  }[]
  videos: { title: string; url: string }[]
  store: string
  categories: string[]

  parent_asin: string
  bought_together: string[] | null
}

File: backend/app/schemas/product.py

from typing import Optional, List, Dict
from pydantic import BaseModel


class Image(BaseModel):
    hi_res: str
    thumb: str
    large: str
    variant: str 

class Video(BaseModel):
    title: str
    url: str    

class Product(BaseModel):
    main_category: str
    title: str
    average_rating: float
    rating_number: float
    features: List[str]
    description: List[str]
    price: float
    images: List[Image]
    videos: List[Video]
    store: str
    categories: List[str]
    parent_asin: str
    bought_together: Optional[List[str]]
